src/mon/NVMeofGwMon.cc

```diff
@@ -211,18 +213,37 @@ void NVMeofGwMon::update_from_paxos(bool *need_bootstrap)
 
 void NVMeofGwMon::check_sub(Subscription *sub)
 {
-  dout(10) << "sub->next , map-epoch " << sub->next
-	   << " " << map.epoch << dendl;
-  if (sub->next <= map.epoch)
-  {
-    dout(10) << "Sending map to subscriber " << sub->session->con
-	     << " " << sub->session->con->get_peer_addr() << dendl;
-    sub->session->con->send_message2(make_message<MNVMeofGwMap>(map));
-
-    if (sub->onetime) {
-      mon.session_map.remove_sub(sub);
-    } else {
-      sub->next = map.epoch + 1;
+//  dout(10) << "sub->next , map-epoch " << sub->next   << " " << map.epoch << dendl;
+  for (auto& created_map_pair: map.created_gws) {
+    auto& group_key = created_map_pair.first;
+    NvmeGwMonStates& gw_created_map = created_map_pair.second;
+    for (auto& gw_created_pair: gw_created_map) {
+      auto& gw_id = gw_created_pair.first;
+      if ( (gw_created_pair.second.availability !=
+          gw_availability_t::GW_CREATED ) &&
```

Comment:
Why filter out GW_CREATED?

Reply:
CREATED  state is not operational state of GW. GW in this state  receives maps via ack-map mechanism . ack-map is sent as response to a beacon.